generator/tree.go

```diff
@@ -160,6 +160,10 @@ func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*
 				return // Inaccessible metrics.
 			}
 
+			if strings.Contains(n.Label, "-") {
+				n.Label = strings.Replace(n.Label, "-", "_", -1)
+			}
+
 			metric := &config.Metric{
 				Name:    n.Label,
```

Comment:
Replace it only down here, changing the actual n.Label could cause issues elsewhere.